src/components/Autocomplete/index.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import domOnlyProps from 'form/domOnlyProps';
+import fieldIsInvalid from 'form/fieldIsInvalid';
+import Autosuggest from 'react-autosuggest';
+import styles from './autocomplete.scss';
+
+const MIN_QUERY_LENGTH = 2;
+
+const Autocomplete = React.createClass({
+  getClassname (element, error) {
+    return [
+      styles[element || 'input'],
+      styles[error ? 'invalid' : 'valid']
+    ].join(' ');
+  },
+
+  handleFetchRequest (e, { endpoint, typeaheadSearch, clearSuggestions }) {
+    // We handle the query length validation here as we don‘t want to do it in
+    // the (throttled) typeahead search. If we did, the first input after
+    // MIN_QUERY_LENGTH would be delayed.
+    let query = e.value;
+
+    if (query) {
+      query = query.trim().toLowerCase();
+
+      if (query.length >= MIN_QUERY_LENGTH) {
+        return typeaheadSearch(endpoint, query);
+      }
+    }
+
+    return clearSuggestions();
+  },
+
+  handleBlur ({
+    value,
+    suggestionFormatter,
+    updateSuggestionInputValue,
+    helper
+  }) {
+    const savedValue = helper.value && suggestionFormatter(helper.value.data);
+
+    // If the saved value in the redux-form matches the displayed value in the
+    // auto-suggest input, we know that the user hasn‘t changed his choice.
+    // If it did, we must reset both values as they aren‘t valid anymore.
+    if (savedValue !== value) {
+      updateSuggestionInputValue('');
+      helper.onChange('');
+    }
+  },
+
+  handleChange (newValue, { updateSuggestionInputValue, helper }) {
+    // We have to manually update the displayed value in the auto-suggest.
+    // This has nothing to do with the actual redux-form value.
+    updateSuggestionInputValue(newValue);
+
+    // If there is no more value, then we know the user has deleted his choice,
+    // and we can update the redux-form store.
+    if (!newValue) {
+      helper.onChange('');
+    }
+  },
+
+  handleSelection (e, { suggestion }, helper) {
+    // Disable form submitting when selecting option with ENTER
+    e.preventDefault();
+
+    // Update redux-form store manually to reflect the selection
+    helper.onChange({
+      data: suggestion,
+      id: suggestion.id,
+      class: 'City'
```

Comment:
this needs to be dynamic. I am not sure how we best pass a data shape - maybe a function as a prop?